core/domain/story_domain_test.py

```diff
@@ -749,6 +792,12 @@ def test_nodes_validation(self) -> None:
         self.story.story_contents.nodes[0].acquired_skill_ids = 1  # type: ignore[assignment]
         self._assert_validation_error(
             'Expected acquired skill ids to be a list, received 1')
+        self.story.story_contents.nodes[0].acquired_skill_ids = ['3']
+        self.assertEqual(
+            len(self.story.story_contents.nodes[0].acquired_skill_ids), 1)
+        self.story.update_node_acquired_skill_ids('node_1', ['3', '4'])
+        self.assertEqual(
+            len(self.story.story_contents.nodes[0].acquired_skill_ids), 2)
```

Comment:
Is there any reason to only check the length rather than asserting that the IDs are `['3', '4']`? You can use `set()` if you're worried about ordering

ditto elsewhere

Reply:
done